libraries/tools/kotlin-gradle-plugin-tcs-android/src/main/kotlin/org/jetbrains/kotlin/gradle/android/AndroidTargetPrototype.kt

```diff
@@ -46,16 +46,48 @@ fun KotlinMultiplatformExtension.androidTargetPrototype(): PrototypeAndroidTarge
     val androidTarget = createExternalKotlinTarget<PrototypeAndroidTarget> {
         targetName = "android"
         platformType = KotlinPlatformType.jvm
-        targetFactory = TargetFactory { delegate ->
-            PrototypeAndroidTarget(delegate, PrototypeAndroidDsl(31))
+        targetFactory = TargetFactory { delegate -> PrototypeAndroidTarget(delegate, PrototypeAndroidDsl(31)) }
+
+        /*
+        Configure apiElements configuration attributes (project to project dependency)
+        In this example we hardcoded AGP version 7.4.0-beta02 as demo
+        */
+        apiElements.configure { _, configuration ->
+            configuration.attributes.attribute(TARGET_JVM_ENVIRONMENT_ATTRIBUTE, project.objects.named(TargetJvmEnvironment.ANDROID))
+            configuration.attributes.attribute(AgpVersionAttr.ATTRIBUTE, project.objects.named("7.4.0-beta02")) /* For demo */
+        }
+
+        /*
+        Configure runtimeElements configuration attributes (project to project dependency)
+        In this example we hardcoded AGP version 7.4.0-beta02 as demo
+        */
+        runtimeElements.configure { _, configuration ->
+            configuration.attributes.attribute(TARGET_JVM_ENVIRONMENT_ATTRIBUTE, project.objects.named(TargetJvmEnvironment.ANDROID))
+            configuration.attributes.attribute(AgpVersionAttr.ATTRIBUTE, project.objects.named("7.4.0-beta02")) /* For demo */
+        }
+
+        /*
+        Configure published configurations (maven publication):
+        We override KotlinPlatformType to be androidJvm for now (to be discussed w/ Google later)
+         */
+        apiElementsPublished.configure { _, configuration ->
+            /* TODO w/ Google: Find a way to deprecate this attribute */
+            configuration.attributes.attribute(KotlinPlatformType.attribute, KotlinPlatformType.androidJvm)
+            configuration.attributes.attribute(TARGET_JVM_ENVIRONMENT_ATTRIBUTE, project.objects.named(TargetJvmEnvironment.ANDROID))
+        }
+
+        runtimeElementsPublished.configure { _, configuration ->
+            /* TODO w/ Google: Find a way to deprecate this attribute */
+            configuration.attributes.attribute(KotlinPlatformType.attribute, KotlinPlatformType.androidJvm)
+            configuration.attributes.attribute(TARGET_JVM_ENVIRONMENT_ATTRIBUTE, project.objects.named(TargetJvmEnvironment.ANDROID))
         }
     }
 
     /*
     Whilst using the .all hook, we only expect the single 'debug' variant to be available through this API.
      */
-    androidExtension.applicationVariants.all { applicationVariant ->
-        project.logger.quiet("Setting up applicationVariant: ${applicationVariant.name}")
+    androidExtension.libraryVariants.all { androidVariant ->
```

Comment:
Note that we should have the api and runtime elements (and published) configurations extend api and runtime classpaths configurations of the main variant. (okay to leave if the aim here is a minimal prototype)

Reply:
I am not yet sure where to put this code, since: Creating the main compilation will be the API consumers (your) job, later. Maybe you should also explicitly do this then?

Reply:
Yes :) I meant that maybe you should add it for the android prototype that you have to be able to test these scenarios. But again it's okay to leave as it as for a minimal prototype.